examples/echo/echo_requester.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a chip-echo-requester, for the
+ *      CHIP Echo Protocol.
+ *
+ *      The CHIP Echo Protocol implements two simple methods, in the
+ *      style of ICMP ECHO REQUEST and ECHO REPLY, in which a sent
+ *      payload is turned around by the responder and echoed back to
+ *      the originator.
+ *
+ */
+
+#include "common.h"
+
+#include <protocols/echo/Echo.h>
+#include <support/ErrorStr.h>
+
+// Max value for the number of EchoRequests sent.
+#define MAX_ECHO_COUNT (3)
```

Comment:
I would prefer `constexpr size_t kMaxEchoCount = 3` inside an anonymous namespace to make use of C++ facilities:

- constexpr so that we have type checking (instead of macros)
- anonymous namespace so we do not use 'static' everywhere. This applies to all static bits below.

Reply:
Acked